### PR TITLE
Fix sentry rejecting legacy request IDs of length larger than 64

### PR DESCRIPTION
**Pull request by JoshVanL**
Fixes https://github.com/dapr/dapr/issues/7078
Proof of fix: https://github.com/dapr/dapr/pull/7081

Fixes an issue whereby sentry would reject SignCertificate requests that have an identifier which was longer than 64 characters. Because sentry still accepts the "legacy" identifier format of `<namespace>:<service account>` (which v1.12 is currently using), Daprd's which are in a namespace and/or using a Service Account with a sufficiently long name will erroneously be rejected.

PR updates the Kubernetes sentry validator so that it will augment the request identifier to the app ID before evaluating whether the identifier is over 64 characters long.

Fix should be back-ported to v1.12 and patch released.

Adds sentry Kubernetes validator integration tests to cover this scenario.

### CODE REVIEW

File: pkg/sentry/sentry.go [hunk 1]
```diff
@@ -158,16 +158,12 @@ func (s *sentry) getValidators(ctx context.Context) (map[sentryv1pb.SignCertific
...
-			if err != nil {
-				return nil, err
-			}
 			sentryID, err := security.SentryID(td, security.CurrentNamespace())
 			if err != nil {
 				return nil, err
 			}
 			val, err := validatorKube.New(ctx, validatorKube.Options{
-				RestConfig:     config,
+				RestConfig:     utils.GetConfig(),
```
Comment (JoshVanL) [inline pkg/sentry/sentry.go:166]: Non-functional change to enable Kubernetes validator integration tests.

File: pkg/sentry/server/ca/ca.go [hunk 2]
```diff
@@ -89,11 +89,7 @@ func New(ctx context.Context, conf config.Config) (Signer, error) {
 	if config.IsKubernetesHosted() {
 		log.Info("Using kubernetes secret store for trust bundle storage")
 
-		restConfig, err := rest.InClusterConfig()
-		if err != nil {
-			return nil, err
-		}
-		client, err := kubernetes.NewForConfig(restConfig)
+		client, err := kubernetes.NewForConfig(utils.GetConfig())
 		if err != nil {
```
Comment (JoshVanL) [inline pkg/sentry/server/ca/ca.go:93]: Non-functional change to enable Kubernetes validator integration tests.

File: pkg/sentry/server/validator/kubernetes/kubernetes.go [hunk 3]
```diff
@@ -192,10 +187,18 @@ func (k *kubernetes) Validate(ctx context.Context, req *sentryv1pb.SignCertifica
...
 		req.Id = expID
 	}
 
+	// The TrustDomain field is ignored by the Kubernetes validator. We must
+	// validate the request _after_ performing the token review so that in the
+	// event the client is uing the "legacy" <ns>:<sa> ID, we can override it
+	// with the expected app ID.
+	if _, _, err = internal.Validate(ctx, req); err != nil {
+		return spiffeid.TrustDomain{}, false, err
+	}
```
Comment (JoshVanL) [inline pkg/sentry/server/validator/kubernetes/kubernetes.go:200]: The fix

File: pkg/sentry/server/validator/kubernetes/kubernetes.go [hunk 4]
```diff
@@ -192,10 +187,18 @@ func (k *kubernetes) Validate(ctx context.Context, req *sentryv1pb.SignCertifica
 	// for the ID containing their namespace and service account (ns:sa). This
 	// is wrong- dapr identities are based on daprd namespace + _app ID_.
 	// Remove this allowance in v1.13.
-	if req.Namespace+":"+pod.Spec.ServiceAccountName == req.Id {
+	if pod.Namespace+":"+pod.Spec.ServiceAccountName == req.Id {
```
Comment (daixiang0) [inline pkg/sentry/server/validator/kubernetes/kubernetes.go:190]: Better to use fmt.Sprintf

File: tests/integration/suite/sentry/validator/insecure/insecure.go [hunk 5]
```diff
@@ -169,3 +171,45 @@ func (m *insecure) Run(t *testing.T, parentCtx context.Context) {
...
+
+func validateCertificateResponse(t *testing.T, res *sentrypbv1.SignCertificateResponse, sentryBundle ca.Bundle, expectSPIFFEID, expectDNSName string) {
+	t.Helper()
+
+	require.NotEmpty(t, res.WorkloadCertificate)
+
+	rest := res.WorkloadCertificate
+
+	// First block should contain the issued workload certificate
+	{
```
Comment (daixiang0) [inline tests/integration/suite/sentry/validator/insecure/insecure.go:183]: No need block, the comment is enough.

File: pkg/sentry/server/validator/kubernetes/kubernetes.go (hunk 4 above)
Comment (JoshVanL) [reply pkg/sentry/server/validator/kubernetes/kubernetes.go:190]: We don't want to use `fmt.Sprintf` here as this is a very hot path for Dapr and bare string concatenation `+` is more performant.

File: tests/integration/suite/sentry/validator/insecure/insecure.go (hunk 5 above)
Comment (JoshVanL) [reply tests/integration/suite/sentry/validator/insecure/insecure.go]: Done

**Review by yaron2 — APPROVED** [review]
LGTM

**Pull request merged**